src/main/java/org/killbill/billing/plugin/analytics/dao/BusinessInvoiceDao.java

```diff
@@ -154,8 +152,12 @@ private void deleteInvoicesAndInvoiceItemsForAccountInTransaction(final Business
 
     private void createInvoiceInTransaction(final BusinessAnalyticsSqlDao transactional,
                                             final BusinessInvoiceModelDao invoice,
-                                            final Iterable<BusinessInvoiceItemBaseModelDao> invoiceItems,
+                                            @Nullable final Iterable<BusinessInvoiceItemBaseModelDao> invoiceItems,
                                             final CallContext context) {
+        if (invoiceItems == null || !InvoiceStatus.COMMITTED.toString().equals(invoice.getStatus())) {
```

Comment:
Should we take into consideration cases where:
* `invoiceItems` != null but invoiceItems.size() == 0 ?
* Other cases for `InvoiceStatus` such as `VOID` ?